### PR TITLE
Add support for linking artifacts from the report

### LINKED ISSUE #2660: New environment variable for link to detailed artifacts
When tests are executed in the Testing Farm it would be nice if `report` plugins would be provided a link to the job artifacts so that the generated report includes the link and user can easily get to the detailed logs. Here are some variable names brainstormed on the stakeholder session:

    TMT_REPORT_ARTIFACTS_URL
    TMT_REPORT_ARTIFACTS_URL_ROOT
    TMT_REPORT_COMMON_ARTIFACTS_URL

Testing Farm would provide this variable to `tmt run` so that the report plugin could pick it. Variable would be provided in the process environment, not as the `tmt run --environment`, for example like this:

    TMT_REPORT_ARTIFACTS_URL=https://artifacts... tmt run ...

In addition to the main link, it would could also useful to provide a direct link to test data directory from each test result. But that can be left for future improvements.

### PR DESCRIPTION
**Pull request by bajertom**
Fixes #2660 

This takes whatever Testing Farm exports as a `TMT_REPORT_ARTIFACTS_URL` environment variable for the tmt run process and puts it in the launch's description field on ReportPortal.

Would it make sense to store it also in `ReportReportPortalData` dataclass? Something like this? 
```
artifacts_url: Optional[str] = field(
metavar="ARTIFACTS_URL",
default=os.getenv('TMT_REPORT_ARTIFACTS_URL'),
help="Launch artifacts link provided by Testing Farm")
```

I suppose `option="--artifacts-url"` is not necessary, because this variable will be provided by TF before tmt runs and user trying to use the option manually on command line will not know the link beforehand anyway.

Pull Request Checklist

* [x] implement the feature
* [x] write the documentation
* [x] extend the test coverage
* [x] modify the json schema
* [x] mention the version
* [x] include a release note

### CODE REVIEW

**Comment by thrix** [comment]
@bajertom it would be great to add some tests for this

File: tmt/steps/report/reportportal.py [hunk 1]
```diff
@@ -318,7 +318,10 @@ def go(self) -> None:
         launch_attributes = self.construct_launch_attributes(suite_per_plan, attributes)
 
         launch_description = self.data.launch_description or self.step.plan.summary
-
+        # Checks whether Testing Farm provided link to artifacts
```
Comment (happz) [inline tmt/steps/report/reportportal.py:321]: I wouldn't tie this to TF in this way - the motivation started in the TF environment, but the concept is generic. The variable can also be set locally, so "Check whether artifact URL has been provided" would serve us better in the long run. One day, tmt might be used by other CI systems directly, who knows?

The variable deserves its paragraph in docs, BTW, somewhere under https://github.com/teemtee/tmt/blob/main/docs/overview.rst?plain=1#L346.

File: tmt/steps/report/reportportal.py [hunk 2]
```diff
@@ -318,7 +318,10 @@ def go(self) -> None:
...
+        if os.getenv('TMT_REPORT_ARTIFACTS_URL'):
+            launch_description = ", ".join(
```
Comment (happz) [inline tmt/steps/report/reportportal.py:323]: 1. f-string would be probably more readable, `launch_description = f"{launch_description}, {os.getenv('TMT_REPORT_ARTIFACTS_URL')}"`
2. `launch_description` might be `None` at this point, and I suspect that `"None, https://artifacts.fo.bar"` would not make a very nice launch description :) How about adding a test for `None` variant?

```python
if not launch_description:
    launch_description = os.getenv('TMT_REPORT_ARTIFACTS_URL')

else:
    launch_description = f"{launch_description}, {os.getenv('TMT_REPORT_ARTIFACTS_URL')}"
```

Comment (KwisatzHaderach) [reply tmt/steps/report/reportportal.py:323]: wouldn't it be better suited for the url to be a launch attribute? maybe we can have it at both places?

File: tmt/steps/report/reportportal.py (hunk 1 above)
Comment (KwisatzHaderach) [reply tmt/steps/report/reportportal.py]: I already created a JIRA issue for myself to do it, but feel free to create an issue here as well. Don't think I'll make it in time for 1.32 as it is supposed to be released this week.

Comment (happz) [reply tmt/steps/report/reportportal.py]: Yeah, probably too late for 1.32 given the number of PRs still in the queue :/

**Comment by thrix** [comment]
@bajertom hi, any update here?

File: tmt/steps/report/reportportal.py (hunk 2 above)
Comment (KwisatzHaderach) [reply tmt/steps/report/reportportal.py]: I would certainly be for that and it seems like an easy change. But if you don't think it makes sense that is fine by me also.

Comment (happz) [reply tmt/steps/report/reportportal.py]: (No preference on my side, I’m here nust for the Python stuff, no idea about RP integration)

**Review by KwisatzHaderach — APPROVED** [review]

File: docs/overview.rst [hunk 3]
```diff
@@ -449,6 +449,11 @@ TMT_REBOOT_TIMEOUT
     How many seconds to wait for a connection to succeed after
     guest reboot. By default, it is 10 minutes.
 
+TMT_REPORT_ARTIFACTS_URL
+    Link to launch artifacts provided for report plugins
```
Comment (psss) [inline docs/overview.rst:453]:
```suggestion
    Link to test artifacts provided for report plugins.
```

As this is not (will not be) related to ReportPortal only?

File: tests/report/reportportal/test.sh [hunk 4]
```diff
@@ -14,6 +14,7 @@ rlJournalStart
     rlPhaseEnd
 
     rlPhaseStartTest
+	rlRun "export REQUESTS_CA_BUNDLE=/etc/pki/tls/certs/ca-bundle.crt"
```
Comment (psss) [inline tests/report/reportportal/test.sh:17]: Hm, I'm not sure we should include this setup in the test. Isn't this rather a user config?

File: tmt/steps/report/reportportal.py [hunk 5]
```diff
@@ -314,10 +314,14 @@ def go(self) -> None:
         attributes = [
             {'key': key, 'value': value[0]}
             for key, value in self.step.plan._fmf_context.items()]
-
         launch_attributes = self.construct_launch_attributes(suite_per_plan, attributes)
 
         launch_description = self.data.launch_description or self.step.plan.summary
+        # Check whether artifact URL has been provided
```
Comment (psss) [inline tmt/steps/report/reportportal.py:320]:
```suggestion
        # Check whether artifacts URL has been provided
```

File: tmt/steps/report/reportportal.py [hunk 6]
```diff
@@ -314,10 +314,14 @@ def go(self) -> None:
...
+        if not launch_description:
+            launch_description = os.getenv('TMT_REPORT_ARTIFACTS_URL')
```
Comment (psss) [inline tmt/steps/report/reportportal.py:322]: What about using a variable instead of repeating `os.getenv()` three times?

**Review by psss — CHANGES_REQUESTED** [review]
Thanks for looking into this. Looks good. Added just a few minor suggestions.

File: tmt/steps/report/reportportal.py (hunk 2 above)
Comment (psss) [reply tmt/steps/report/reportportal.py]: Ack, if it's not clickable, let's not add it to attributes.

File: tmt/steps/report/reportportal.py (hunk 6 above)
Comment (psss) [reply tmt/steps/report/reportportal.py]: Good, what about including the option as well?

File: tests/report/reportportal/test.sh (hunk 4 above)
Comment (psss) [reply tests/report/reportportal/test.sh]: Thanks.

File: tmt/steps/report/reportportal.py (hunk 6 above)
Comment (happz) [reply tmt/steps/report/reportportal.py]: That is already the option, as `self.data.artifacts_url` is populated from fmf key, CLI option, or envvar.

Comment (psss) [reply tmt/steps/report/reportportal.py]: I meant 36a03b0887829ad8ecaded5c5d48a7b2e33038fd.

**Comment by psss** [comment]
Thanks for the adjustments. I've updated the pull request checklist: We need to modify the json schema as well and I think it's worth to include the new variable in the release notes.

**Review by psss — APPROVED** [review]
Thanks for all the changes, now looks good!
